fmn/consumer/backend.py

```diff
@@ -175,9 +178,11 @@ def read(queue_object):
                 idx = recipient['filter_id']
                 fltr = session.query(fmn.lib.models.Filter).get(idx)
                 fltr.fired(session)
-        except:
-            # If anything happens, put the message back in the queue and bail
-            ch.cancel()
+        except RuntimeError:
+            yield ch.basic_nack(delivery_tag=method.delivery_tag)
+            return
+        except smtplib.SMTPSenderRefused:
```

Comment:
So you are right that it's definitely not ideal, but with the way the code was, the user would also not be getting the email while the FMN queue would break and stop processing the messages (or end up in an infinite loop).
So this is a little bit of a band-aid on a wooden leg, it fixes a symptom but not really the root cause which is that the message is too big.

As for splitting the email to actually fix the symptom I really do not know how feasible that is. We have seen the issue happening but we do not know which filter created it. Was it a batch of too many messages? A very large commit in dist-git with a lot of files and therefore a huge diff to send?
If it was a batch that's too big, we may be able to "fix" it by reducing its size, if it is a diff then I don't really know how we could achieve it.

So yes, there is room for improvement there but it's a little bit of a corner case that we do not exactly know how to reproduce and with these changes at least we got the consumer running again :)